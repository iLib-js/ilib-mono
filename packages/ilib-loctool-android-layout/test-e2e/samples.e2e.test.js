/*
 * samples.e2e.test.js - E2E tests for ilib-loctool-android-layout samples
 *
 * Copyright Â© 2025, Box, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

const path = require("path");
const fs = require("fs");
const { expectFileToMatchSnapshot, LoctoolRunner } = require("ilib-internal");

describe("samples", () => {
    describe("android", () => {
        const projectPath = path.resolve(__dirname, "..", "samples", "android");
        const xliffPath = path.resolve(projectPath, "sample-extracted.xliff");

        beforeAll(async () => {
            const loctool = new LoctoolRunner(projectPath);
            await loctool.run("localize");
        });

        afterAll(() => {
            // Clean up all files generated by loctool localize
            const filesToClean = [
                xliffPath, // sample-extracted.xliff
                path.resolve(projectPath, "sample-new-ko-KR.xliff"),
                path.resolve(projectPath, "sample-new-ru-RU.xliff"),
                path.resolve(projectPath, "res/values/strings-auto.xml")
            ];

            filesToClean.forEach(file => {
                if (fs.existsSync(file)) {
                    fs.unlinkSync(file);
                }
            });

            // Clean up generated directories
            const dirsToClean = [
                path.resolve(projectPath, "res/values-ko"),
                path.resolve(projectPath, "res/values-ru")
            ];

            dirsToClean.forEach(dir => {
                if (fs.existsSync(dir)) {
                    fs.rmSync(dir, { recursive: true, force: true });
                }
            });
        });

        it("should produce an extracted XLIFF file", () => {
            expectFileToMatchSnapshot(xliffPath);
        });
    });
});
