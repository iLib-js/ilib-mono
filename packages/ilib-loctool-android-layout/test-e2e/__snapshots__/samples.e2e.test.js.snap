// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`samples android should produce an extracted XLIFF file 1`] = `
"<?xml version="1.0" encoding="utf-8"?>
<xliff version="1.2">
  <file original="res/layout/t1.xml" source-language="en-US" product-name="sample">
    <body>
      <trans-unit id="22" resname="text_Unlimited_Gigabytes_of_Data" restype="string" datatype="x-android-resource">
        <source>Unlimited Gigabytes of Data</source>
      </trans-unit>
      <trans-unit id="23" resname="text_Choose" restype="string" datatype="x-android-resource">
        <source>Choose</source>
      </trans-unit>
    </body>
  </file>
  <file original="res/values/strings.xml" source-language="en-US" product-name="sample">
    <body>
      <trans-unit id="1" resname="app_id" restype="string" datatype="x-android-resource">
        <source>151779581544891</source>
        <note>Do not translate</note>
      </trans-unit>
      <trans-unit id="2" resname="disclaimer" restype="string" datatype="x-android-resource">
        <source>Disclaimer</source>
      </trans-unit>
      <trans-unit id="3" resname="description_imgVw" restype="string" datatype="x-android-resource">
        <source>imageView</source>
      </trans-unit>
      <trans-unit id="4" resname="thanks_friend_pre" restype="string" datatype="x-android-resource">
        <source>Send a thank you note to\\n{name}</source>
      </trans-unit>
      <trans-unit id="5" resname="thanks_news" restype="string" datatype="x-android-resource">
        <source>{name} appreciates your gratitude :)</source>
      </trans-unit>
      <trans-unit id="6" resname="thanks" restype="string" datatype="x-android-resource">
        <source>Thank you!</source>
      </trans-unit>
      <trans-unit id="7" resname="talk_to_friend" restype="string" datatype="x-android-resource">
        <source>Get help now</source>
      </trans-unit>
      <trans-unit id="8" resname="ask_question" restype="string" datatype="x-android-resource">
        <source>Ask friends</source>
      </trans-unit>
      <trans-unit id="9" resname="thank" restype="string" datatype="x-android-resource">
        <source>Thank</source>
      </trans-unit>
      <trans-unit id="10" resname="thanked" restype="string" datatype="x-android-resource">
        <source>Thanked</source>
      </trans-unit>
      <trans-unit id="11" resname="thanked_note" restype="string" datatype="x-android-resource">
        <source>Leave a note?</source>
      </trans-unit>
      <trans-unit id="12" resname="r707551335" restype="string" datatype="x-android-resource">
        <source>{meeting_time} with {friend_name}</source>
      </trans-unit>
      <trans-unit id="13" resname="days" restype="array" datatype="x-android-resource" extype="0">
        <source>Sunday</source>
      </trans-unit>
      <trans-unit id="14" resname="days" restype="array" datatype="x-android-resource" extype="1">
        <source>Monday</source>
      </trans-unit>
      <trans-unit id="15" resname="days" restype="array" datatype="x-android-resource" extype="2">
        <source>Tuesday</source>
      </trans-unit>
      <trans-unit id="16" resname="days" restype="array" datatype="x-android-resource" extype="3">
        <source>Wednesday</source>
      </trans-unit>
      <trans-unit id="17" resname="days" restype="array" datatype="x-android-resource" extype="4">
        <source>Thursday</source>
      </trans-unit>
      <trans-unit id="18" resname="days" restype="array" datatype="x-android-resource" extype="5">
        <source>Friday</source>
      </trans-unit>
      <trans-unit id="19" resname="days" restype="array" datatype="x-android-resource" extype="6">
        <source>Saturday</source>
      </trans-unit>
      <trans-unit id="20" resname="app_recommend2" restype="plural" datatype="x-android-resource" extype="one">
        <source>{start}1 friend{end} recommends this app</source>
        <note>Make sure {start} and {end} are surrounding the translated words of '1 friend' or '%d friends'.</note>
      </trans-unit>
      <trans-unit id="21" resname="app_recommend2" restype="plural" datatype="x-android-resource" extype="other">
        <source>{start}%d friends{end} recommend this app</source>
        <note>Make sure {start} and {end} are surrounding the translated words of '1 friend' or '%d friends'.</note>
      </trans-unit>
      <trans-unit id="24" resname="app_id" restype="string" datatype="x-android-resource">
        <source>151779581544891</source>
        <note>Do not translate</note>
      </trans-unit>
      <trans-unit id="25" resname="disclaimer" restype="string" datatype="x-android-resource">
        <source>Disclaimer</source>
      </trans-unit>
      <trans-unit id="26" resname="description_imgVw" restype="string" datatype="x-android-resource">
        <source>imageView</source>
      </trans-unit>
      <trans-unit id="27" resname="thanks_friend_pre" restype="string" datatype="x-android-resource">
        <source>Send a thank you note to\\n{name}</source>
      </trans-unit>
      <trans-unit id="28" resname="thanks_news" restype="string" datatype="x-android-resource">
        <source>{name} appreciates your gratitude :)</source>
      </trans-unit>
      <trans-unit id="29" resname="thanks" restype="string" datatype="x-android-resource">
        <source>Thank you!</source>
      </trans-unit>
      <trans-unit id="30" resname="talk_to_friend" restype="string" datatype="x-android-resource">
        <source>Get help now</source>
      </trans-unit>
      <trans-unit id="31" resname="ask_question" restype="string" datatype="x-android-resource">
        <source>Ask friends</source>
      </trans-unit>
      <trans-unit id="32" resname="thank" restype="string" datatype="x-android-resource">
        <source>Thank</source>
      </trans-unit>
      <trans-unit id="33" resname="thanked" restype="string" datatype="x-android-resource">
        <source>Thanked</source>
      </trans-unit>
      <trans-unit id="34" resname="thanked_note" restype="string" datatype="x-android-resource">
        <source>Leave a note?</source>
      </trans-unit>
      <trans-unit id="35" resname="r707551335" restype="string" datatype="x-android-resource">
        <source>{meeting_time} with {friend_name}</source>
      </trans-unit>
      <trans-unit id="36" resname="days" restype="array" datatype="x-android-resource" extype="0">
        <source>Sunday</source>
      </trans-unit>
      <trans-unit id="37" resname="days" restype="array" datatype="x-android-resource" extype="1">
        <source>Monday</source>
      </trans-unit>
      <trans-unit id="38" resname="days" restype="array" datatype="x-android-resource" extype="2">
        <source>Tuesday</source>
      </trans-unit>
      <trans-unit id="39" resname="days" restype="array" datatype="x-android-resource" extype="3">
        <source>Wednesday</source>
      </trans-unit>
      <trans-unit id="40" resname="days" restype="array" datatype="x-android-resource" extype="4">
        <source>Thursday</source>
      </trans-unit>
      <trans-unit id="41" resname="days" restype="array" datatype="x-android-resource" extype="5">
        <source>Friday</source>
      </trans-unit>
      <trans-unit id="42" resname="days" restype="array" datatype="x-android-resource" extype="6">
        <source>Saturday</source>
      </trans-unit>
      <trans-unit id="43" resname="app_recommend2" restype="plural" datatype="x-android-resource" extype="one">
        <source>{start}1 friend{end} recommends this app</source>
        <note>Make sure {start} and {end} are surrounding the translated words of '1 friend' or '%d friends'.</note>
      </trans-unit>
      <trans-unit id="44" resname="app_recommend2" restype="plural" datatype="x-android-resource" extype="other">
        <source>{start}%d friends{end} recommend this app</source>
        <note>Make sure {start} and {end} are surrounding the translated words of '1 friend' or '%d friends'.</note>
      </trans-unit>
    </body>
  </file>
</xliff>"
`;
